### ISSUE TITLE
segment selectedSegmentIndex not working

### ISSUE THREAD

**Issue by syedrazackimran** [GocePetrovski/ScrollableSegmentedControl#4]
hi 

`@IBOutlet var segment: ScrollableSegmentedControl!
override func viewDidLoad() {
 super.viewDidLoad()
let largerRedTextAttributes = [NSFontAttributeName: UIFont(name: "Avenir-Roman", size: 12)!,
                                       NSForegroundColorAttributeName: #colorLiteral(red: 0.9411764706, green: 0.9411764706, blue: 0.9411764706, alpha: 1)] as [String : Any]
        
        let largerRedTextHighlightAttributes = [NSFontAttributeName: UIFont(name: "Avenir-Roman", size: 12)!,
                                                NSForegroundColorAttributeName: #colorLiteral(red: 0.9411764706, green: 0.9411764706, blue: 0.9411764706, alpha: 1)] as [String : Any]
        
        let largerRedTextSelectAttributes = [NSFontAttributeName: UIFont(name: "Avenir-Medium", size: 14)!,
                                             NSForegroundColorAttributeName: ApplicationColor.BlueColor]

        // change some colors
        segment.segmentContentColor = #colorLiteral(red: 0.9411764706, green: 0.9411764706, blue: 0.9411764706, alpha: 1)
        segment.selectedSegmentContentColor = ApplicationColor.BlueColor
        segment.backgroundColor = .white
        segment.segmentStyle = .imageOnTop
        segment.insertSegment(withTitle: "Segment 1", image: #imageLiteral(resourceName: "male"), at: 0)
        segment.insertSegment(withTitle: "Segment 2", image: #imageLiteral(resourceName: "female"), at: 1)
        segment.insertSegment(withTitle: "Segment 3", image: #imageLiteral(resourceName: "price"), at: 2)
        segment.insertSegment(withTitle: "Segment 4", image: #imageLiteral(resourceName: "time"), at: 3)
        segment.insertSegment(withTitle: "Segment 5", image: #imageLiteral(resourceName: "cellphone"), at: 4)
        segment.insertSegment(withTitle: "Segment 6", image: #imageLiteral(resourceName: "profile"), at: 5)
        segment.underlineSelected = true
        segment.selectedSegmentIndex = 4
        segment.setTitleTextAttributes(largerRedTextAttributes, for: .normal)
        segment.setTitleTextAttributes(largerRedTextHighlightAttributes, for: .highlighted)
        segment.setTitleTextAttributes(largerRedTextSelectAttributes, for: .selected) 
}
 @IBAction func segmentChanged(_ sender: ScrollableSegmentedControl) {
        print("Segment at index \(sender.selectedSegmentIndex)  selected")

    }`

> Segment at index 4  selected
Segment at index -1  selected
Segment at index 0  selected

**above mentioned is my code to create segment! I need to selected index as 4 but when is loaded console log return different index, it's from -(minus) index, kindly check it and help to resolve it **

**Comment by GocePetrovski** [comment]
Hi,
I will fix this issue in the coming days. 
As workaround for now you can set the selected segment index in ``` viewDidAppear ``` instead. 

Example:
```
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        segmentedControl.selectedSegmentIndex = 4
    }
```

**Comment by GocePetrovski** [comment]
Actually it was easier to fix than I thought, so I did it and published version [1.2.1](https://github.com/GocePetrovski/ScrollableSegmentedControl/releases/tag/1.2.1)

Closing this issue.
